Ginger/GingerCore/ALM/QtestCore.cs

```diff
@@ -0,0 +1,661 @@
+
+                                                if (!System.IO.File.Exists(zipFileName))
+                                                {
+                                                    ZipFile.CreateFromDirectory(activGroup.TempReportFolder, zipFileName);
+                                                }
+                                                else
+                                                {
+                                                    System.IO.File.Delete(zipFileName);
+                                                    ZipFile.CreateFromDirectory(activGroup.TempReportFolder, zipFileName);
+                                                }
+                                                System.IO.Directory.Delete(activGroup.TempReportFolder, true);
+                                                // to discuss an issue
+                                                attachmentApi.Upload((long)Convert.ToInt32(ALMCore.DefaultAlmConfig.ALMProjectKey), "test-logs", testLog.Id, "GingerExecutionHTMLReport.zip", "application/x-zip-compressed", File.ReadAllBytes(zipFileName));
+                                                System.IO.File.Delete(zipFileName);
+                                            }
+                                        }
+                                    }                                   
+                                }
+                                else
+                                {
+                                    //No matching TC was found for the ActivitiesGroup in QC
+                                    result = "Matching TC's were not found for all " + GingerDicser.GetTermResValue(eTermResKey.ActivitiesGroups) + " in qTest.";
+                                }
+                            }
+                        }
+                    }
+                    else
+                    {
+                        //No matching Test Set was found for the BF in QC
+                        result = "No matching Test Set was found in qTest.";
+                    }
+
+                }
+                if (result == string.Empty)
+                {
+                    result = "Export performed successfully.";
+                    return true;
+                }
+                else
+                {
+                    return false;
+                }
+            }
+            catch (Exception ex)
+            {
+                result = "Unexpected error occurred- " + ex.Message;
+                Reporter.ToLog(eLogLevel.ERROR, "Failed to export execution details to Qtest", ex);               
+                return false;
+            }
+        }
+
+        public void SetTestStepLogStatus(Activity matchingActivity, ref QTestApiModel.TestStepLogResource testStepLog, List<QTestApiModel.StatusResource> statuses)
+        {
+            if (matchingActivity != null)
+            {
+                switch (matchingActivity.Status)
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Add a 'default' clause to this 'switch' statement.](https://app.codacy.com/manual/Ginger-Automation/Ginger/pullRequest?prid=5395796)